programs/client/QueryFuzzer.cpp

```diff
@@ -305,6 +306,26 @@ void QueryFuzzer::fuzz(ASTPtr & ast)
             replaceWithColumnLike(ast);
         }
     }
+    else if (auto * dictionary_layout = typeid_cast<ASTDictionaryLayout *>(ast.get()))
+    {
+        // Don't fuzz it. It is a somewhat unconventional AST which does not
+        // directly correspond to the query text, but performs some validation
+        // by itself. It considers its children to be ASTLiteral * only, so when
+        // we replace them with something column-like, the client segfaults.
```

Comment:
But it should not crash, why don't just throw exception about wrong AST?

Reply:
That is the problem -- the AST is validated at the parsing stage, so if you change it after that, you get a segfault. It is not possible to trigger from SQL. Normally we keep the AST close to what is written, and validate it later -- this is the only case I encountered so far that doesn't work like this.

Reply:
Why don't add yet another validation even if it is not needed at all in normal runs?